pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -91,8 +91,8 @@ def advance(
         if batch is None:
             raise StopIteration
 
-        with self.trainer.profiler.profile("evaluation_batch_to_device"):
-            batch = self.trainer.accelerator.batch_to_device(batch, dataloader_idx=dataloader_idx)
+        if not self.trainer.data_connector.evaluation_data_fetcher.store_on_device:
+            batch = self.trainer.accelerator.batch_to_device(batch)
```

Comment:
What about the prediction loop? Do we need to care for that as well?

Reply:
It is not enable right now, but might come in future PRs :)